File: internal/models/review.go
package models

import "time"

type Review struct {
	ID            int       `db:"id" json:"id"`
	ListingID     int       `db:"listing_id" json:"listing_id"`
	ReviewerName  string    `db:"reviewer_name" json:"reviewer_name"`
	ReviewerEmail *string   `db:"reviewer_email" json:"reviewer_email"`
	Rating        int       `db:"rating" json:"rating"`
	Comment       *string   `db:"comment" json:"comment"`
	IsApproved    bool      `db:"is_approved" json:"is_approved"`
	CreatedAt     time.Time `db:"created_at" json:"created_at"`
}

File: mobile/lib/models/review.dart
class Review {
  final int id;
  final int listingId;
  final String reviewerName;
  final String? reviewerEmail;
  final int rating;
  final String? comment;
  final bool isApproved;
  final String createdAt;

  Review({
    required this.id,
    required this.listingId,
    required this.reviewerName,
    this.reviewerEmail,
    required this.rating,
    this.comment,
    required this.isApproved,
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'] ?? 0,
      listingId: json['listing_id'] ?? 0,
      reviewerName: json['reviewer_name'] ?? '',
      reviewerEmail: json['reviewer_email'],
      rating: json['rating'] ?? 0,
      comment: json['comment'],
      isApproved: json['is_approved'] ?? false,
      createdAt: json['created_at'] ?? '',
    );
  }
}
